TLC59116_Unmanaged.cpp

```diff
@@ -187,16 +190,12 @@ byte TLC59116_Unmanaged::best_iref(byte ma, int Rext) {
       }
     if (ma >= 3150/Rext) {
       if (ma <= 9449.98/Rext) {
-        HC=1; CM=0; D = (1.280*ma*Rext )/63-64;
-        byte proposed = i_out_d(CM, HC, D, Rext);
+        HCp=1; CMp=0; Dp = (1.280*ma*Rext )/63-64;
+        byte proposed = i_out_d(CMp, HCp, Dp, Rext);
         // The solutions overlap, so:
-        if ( proposed >= actual) actual=proposed;
-        else {
-          // printf("  tried proposed %dma HC %d, CM %d, D %03o but, use previous segment\n", proposed, HC, CM, D);
-          // revert
-          D=D*2; // 1.28 -> 2.56
-          HC=0;
-          actual = i_out_d(CM, HC, D, Rext);
+        if ( proposed >= actual) {
+		  actual = proposed;
+	      HC=HCp; CM=CMp; D=Dp;
```

Comment:
(note to self: check HC use: saved? why "p")

Reply:
For proposed.    HC"proposed".     Otherwise I was seeing a condition where it was clobbering HC and no reverting it.   When it exited the loop, HC was incorrect and setpoint was off.   Rather than revert, I just decided to never alter the original value in the first place.